src/server/serverBase.ts

```diff
@@ -103,12 +103,12 @@ const findMatchingFile = (filePath: string, fileNames: string[], expectedFileNam
                 }
             }
             catch {
-                errorsCaught ++; // swallow error and continue the loop
+                filesWithParsingError.push(fileName); // swallow error and continue the loop
             }
         }
     }
-    if (errorsCaught > 0) {
-        throw new Error();
+    if (filesWithParsingError.length > 0) {
+        throw new Error(filesWithParsingError.join(','));
```

Comment:
Really minor nit - since this error is surfaced to the user, I think we should add a space to the join string so it appears as `a.json, b.json` instead of `a.json,b.json`